crates/cli-support/src/js/mod.rs

```diff
@@ -2634,63 +2635,115 @@ impl<'a, 'b> SubContext<'a, 'b> {
+                    "",
+                    &format!("wasm.{}", wasm_name),
+                    ExportedShim::Named(&wasm_name),
+                ))
+        };
+        match &export.method_kind {
+            decode::MethodKind::Constructor => {
+                fn_name = "constructor";
+                let mut j2r = Js2Rust::new(fn_name, self.cx);
+                j2r.constructor(Some(class_name));
+                let (js, ts, raw_docs) = finish_j2r(j2r)?;
+                let class = require_class(&mut self.cx.exported_classes, class_name);
+                if class.has_constructor {
+                    bail!("found duplicate constructor `{}`", export.function.name);
+                }
+                class.has_constructor = true;
+                let docs = docs(raw_docs);
+                method(class, &docs, fn_name, "", &js, &ts);
+                Ok(())
+            }
+            decode::MethodKind::Operation(operation) => {
+                let mut j2r = Js2Rust::new(fn_name, self.cx);
+                let mut static_prfx = "";
+                if operation.is_static {
+                    static_prfx = "static ";
+                } else {
+                    j2r.method(export.consumed);
+                }
+                let (js, ts, raw_docs) = finish_j2r(j2r)?;
+                let class = require_class(&mut self.cx.exported_classes, class_name);
+                let docs = docs(raw_docs);
+                match operation.kind {
+                    decode::OperationKind::Getter(maybe_getter_name) => {
+                        if export.function.renamed_via_js_name {
+                            method(class, &docs, fn_name, "", &js, &ts);
+                            getter_from_method_call(class, &docs, fn_name, maybe_getter_name);
+                        } else {
+                            let getter_name = if maybe_getter_name.is_empty() {
+                                fn_name
+                            } else {
+                                maybe_getter_name
+                            };
+                            method(class, &docs, getter_name, "get ", &js, &ts);
+                        }
+                    }
+                    decode::OperationKind::Setter(maybe_setter_name) => {
+                        if export.function.renamed_via_js_name {
+                            method(class, &docs, fn_name, "", &js, &ts);
+                            setter_from_method_call(class, &docs, fn_name, &js, maybe_setter_name);
+                        } else {
+                            let getter_name = if maybe_setter_name.is_empty() {
+                                fn_name
+                            } else {
+                                maybe_setter_name
+                            };
+                            method(class, &docs, getter_name, "set ", &js, &ts);
+                        }
+                    }
+                    _ => method(class, &docs, fn_name, static_prfx, &js, &ts),
```

Comment:
Should an error be thrown? This line code roughly translates to: Don't do fancy things for anything that isn't a getter or a setter.

Reply:
Ah yeah I think we should generate an error here because something like `#[wasm_bindgen(indexing_getter)]` (or something like that) isn't fully supported here